Libbna/CUSTOM-CMS
# #221: Search feature Implementation.

* Added search template.
* Added controller and model methods that handle the search request.
* Displayed the search results on search template..

* Testing steps
    * Type the query in search input field.
    * Hit enter or click the search icon.
    * Results or no results page should get rendered.
    * This search works for misspelled queries as well.
    
    * Also, Go to article page, scroll down and click on any category in the category.
    * Articles related to the category should be listed as a result.
